Applications/Utils/SimpleMeshCreation/createMeshElemPropertiesFromASCRaster.cpp

```diff
@@ -35,212 +36,158 @@
-            "factor (default = 1)");
-    cmd.add( refinement_arg );
-
-    TCLAP::ValueArg<std::string> mapping_arg("",
-                                             "mapping-name",
-                                             "file name of mapping",
-                                             true,
-                                             "",
-                                             "file name");
-    cmd.add( mapping_arg );
+        "r",
+        "refine",
+        "refinement factor that raises the resolution of the raster data",
+        false,
+        1,
+        "factor (default = 1)");
+    cmd.add(refinement_arg);
 
     TCLAP::ValueArg<std::string> raster_arg("",
                                             "raster-file",
                                             "the name of the ASC raster file",
                                             true,
                                             "",
                                             "file name");
-    cmd.add( raster_arg );
+    cmd.add(raster_arg);
+
+    TCLAP::ValueArg<std::string> property_arg(
+        "p",
+        "property-name",
+        "the name of the property the values are stored for",
+        false,
+        "gwn",
+        "property name as string");
+    cmd.add(property_arg);
 
     TCLAP::ValueArg<std::string> mesh_arg("m",
                                           "mesh",
                                           "the mesh is read from this file",
                                           true,
                                           "test.msh",
                                           "file name");
-    cmd.add( mesh_arg );
+    cmd.add(mesh_arg);
+
+    std::vector<std::string> allowed_units{ "mm/a", "mm/month", "m/s" };
+    TCLAP::ValuesConstraint<std::string> allowed_units_constraints{
+        allowed_units};
+    TCLAP::ValueArg<std::string> unit_arg("u",
+                                          "input-unit",
+                                          "input unit of the data",
+                                          true,
+                                          "m/s",
+                                          &allowed_units_constraints);
+    cmd.add(unit_arg);
 
-    cmd.parse( argc, argv );
+    cmd.parse(argc, argv);
 
     // read mesh
-    std::unique_ptr<MeshLib::Mesh> dest_mesh(MeshLib::IO::readMeshFromFile(mesh_arg.getValue()));
+    std::unique_ptr<MeshLib::Mesh> dest_mesh(
+        MeshLib::IO::readMeshFromFile(mesh_arg.getValue()));
 
     // read raster and if required manipulate it
     auto raster = std::unique_ptr<GeoLib::Raster>(
-        FileIO::AsciiRasterInterface::getRasterFromASCFile(raster_arg.getValue()));
-    GeoLib::RasterHeader header (raster->getHeader());
-    if (refinement_arg.getValue() > 1) {
+        FileIO::AsciiRasterInterface::getRasterFromASCFile(
+            raster_arg.getValue()));
+    GeoLib::RasterHeader header(raster->getHeader());
+    if (refinement_arg.getValue() > 1)
```

Comment:
To coarsen the data means information loss. The additional refinement parameter should give the user the possibility to refine the raster such that each cell of the destination mesh contains at least some (raster) information.